src/test/unit/test_lumigo_utils.py

```diff
@@ -411,22 +415,70 @@ def test_report_json_china_missing_secret_access_key(monkeypatch, reporter_mock,
     )
 
 
-def test_report_json_china_no_boto(monkeypatch, reporter_mock, caplog):
+@pytest.mark.parametrize("package_name", ["boto3", "botocore"])
+def test_report_json_china_no_boto(package_name, monkeypatch, reporter_mock, caplog):
     reporter_mock.side_effect = report_json
     monkeypatch.setattr(Configuration, "should_report", True)
     monkeypatch.setattr(Configuration, "edge_kinesis_aws_access_key_id", "my_value")
     monkeypatch.setattr(Configuration, "edge_kinesis_aws_secret_access_key", "my_value")
-    monkeypatch.setattr(lumigo_utils, "boto3", None)
+    monkeypatch.setattr(lumigo_utils, package_name, None)
 
     report_json(CHINA_REGION, [{"a": "b"}])
 
     assert any(
-        "boto3 is missing. Unable to send to Kinesis" in record.message
+        f"{package_name} is missing. Unable to send to Kinesis" in record.message
         and record.levelname == "ERROR"  # noqa
         for record in caplog.records
     )
 
 
+def test_report_json_china_on_error_no_exception_and_notify_user(capsys, monkeypatch):
+    monkeypatch.setattr(Configuration, "should_report", True)
+    monkeypatch.setattr(Configuration, "edge_kinesis_aws_access_key_id", "my_value")
+    monkeypatch.setattr(Configuration, "edge_kinesis_aws_secret_access_key", "my_value")
+    monkeypatch.setattr(boto3, "client", MagicMock(side_effect=Exception))
+    lumigo_utils.get_logger().setLevel(logging.CRITICAL)
+
+    report_json(CHINA_REGION, [{"a": "b"}])
+
+    assert "Failed to send spans" in capsys.readouterr().out
+
+
+def test_china_shouldnt_establish_http_connection(monkeypatch):
+    monkeypatch.setenv("AWS_REGION", CHINA_REGION)
+    # Reload a duplicate of lumigo_utils
+    spec = importlib.util.find_spec("lumigo_tracer.lumigo_utils")
```

Comment:
This test has an impact on the other tests - they'll have this new `lumigo_tracer.lumigo_utils` module.
Moreover, checkout `test_no_wrapping_library` for an easier way to do it